tests/basic/pure_test.go

```diff
@@ -2365,3 +2364,279 @@ var _ = Describe("{FADAVolMigrateValidation}", func() {
+	})
+
+	var (
+		hostName               = "torpedo-host"
+		volumeName             = "torpedo-vol"
+		faSecret               = Inst().FaSecret
+		FAclient               *flasharray.Client
+		MultipathBeforeRestart string
+		faMgmtEndPoint         string
+		faAPIToken             string
+	)
+
+	itLog := "Attach a volume from a different FA, restart portworx and check multipath consistency"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		// select a random node to run the test
+		n := node.GetStorageDriverNodes()[0]
+
+		stepLog := "get the secrete of FA which is not present in pure secret"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			//get the flash array details
+			volDriverNamespace, err := Inst().V.GetVolumeDriverNamespace()
+			log.FailOnError(err, "failed to get volume driver [%s] namespace", Inst().V.String())
+
+			pxPureSecret, err := pureutils.GetPXPureSecret(volDriverNamespace)
+			log.FailOnError(err, "Failed to get secret %v", pxPureSecret)
+			flashArraysInSecret := pxPureSecret.Arrays
+
+			if len(flashArraysInSecret) == 0 {
+				log.FailOnError(fmt.Errorf("no FlashArrays details found"), fmt.Sprintf("error getting FlashArrays creds from %s [%s]", PureSecretName, pxPureSecret))
+			}
+
+			for _, value := range strings.Split(faSecret, ",") {
+				faMgmtEndPoint = strings.Split(value, ":")[0]
+				faAPIToken = strings.Split(value, ":")[1]
+				if len(faMgmtEndPoint) == 0 || len(faAPIToken) == 0 {
+					continue
+				}
+				log.InfoD("famanagement endpoint: %v, faAPIToken: %v", faMgmtEndPoint, faAPIToken)
+				break
+			}
+			if len(faMgmtEndPoint) == 0 || len(faAPIToken) == 0 {
+				log.FailOnError(fmt.Errorf("no FlashArrays details found"), fmt.Sprintf("error getting FlashArrays creds from %s [%s]", PureSecretName, pxPureSecret))
+			}
+
+			for _, fa := range flashArraysInSecret {
+				if fa.MgmtEndPoint == faMgmtEndPoint {
+					log.FailOnError(fmt.Errorf("Flash Array details present in secret"), "Flash Array details should not be present in the secret")
+				}
+			}
+		})
+
+		stepLog = "Create a volume, create a host, attach the volume to the host, update iqn of the host and attach the volume to the host"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			iqn, err := GetIQNOfNode(n)
+			log.FailOnError(err, "Failed to get iqn of the node %v", n.Name)
+			log.InfoD("Iqn of the node: %v", iqn)
+
+			//create a connections to the FA
+			FAclient, err = pureutils.PureCreateClientAndConnect(faMgmtEndPoint, faAPIToken)
+			log.FailOnError(err, "Failed to create client and connect to FA")
+
+			//create a volume on the FA
+			volume, err := pureutils.CreateVolumeOnFABackend(FAclient, volumeName, 1048576)
+			log.FailOnError(err, "Failed to create volume on FA")
+			log.InfoD("Volume created on FA: %v", volume.Name)
+
+			//create a host in the FA
+			host, err := pureutils.CreateNewHostOnFA(FAclient, hostName)
+			log.FailOnError(err, "Failed to create host on FA")
+			log.InfoD("Host created on FA: %v", host.Name)
+
```

Comment:
Validate if IQN is already present on some hosts

Reply:
Done. If Iqn is present taking the host which is already present.